### PR TITLE
DOC: Complete adding docstrings to Pydantic models

### LINKED ISSUE #696: Update `datamodel.rst` documentation
There are a few things we can update here.

- It is in general writing about the legacy schema. Much of this still applies, but some details have changed
- The **Full schema** embed should point to the new schema -- although given its length, it might not be very readable on its own.
- ~`data.geometry` should be added~
- #695 
- Much of the documentation in this page could and should be copied as docstrings to the relevant Pydantic models if it's not been done already

If we can coerce the Pydantic autodoc into something more readable and nested this document may become redundant. Alternatively, it may be possible to autogenerate a similar page, without Pydantic implementation details, from the same autodoc.

### PR DESCRIPTION
**Pull request by mferrera**
Resolves #696 

Also adds them to enums, and renames some classes to remove the redundant `Enum` in the name or align with other names (i.e. FMUSomething as opposed to FmuSomething).

Removing descriptions from the `Field(description="...")` also removes these descriptions from the schema right now. However, with Pydantic 2.7 (when we are able to upgrade to it), moving these descriptions to docstrings will add them back into the schema description. So this loss of fidelity is temporary.

### CODE REVIEW

File: src/fmu/dataio/datastructure/meta/content.py [hunk 1]
```diff
@@ -20,65 +20,72 @@
...
+    """The filter size applied."""
+
+    scaling_factor: Optional[float] = Field(allow_inf_nan=False, default=None)
+    """The scaling factor applied."""
+
+    stacking_offset: Optional[str] = Field(default=None, examples=["0-15"])
+    """The stacking offset applied."""
+
+    zrange: Optional[float] = Field(allow_inf_nan=False, default=None)
+    """The z-rage of these data."""
```
Comment (tnatt) [inline src/fmu/dataio/datastructure/meta/content.py:74]:
```suggestion
    """The z-range of these data."""
```

File: src/fmu/dataio/datastructure/meta/content.py [hunk 2]
```diff
@@ -95,29 +102,44 @@ def contact_to_lowercase(cls, v: str) -> str:
...
+    """
+    The geometry of the object, i.e. the grid that an object representing a grid
+    property is derivative of.
+    """
+
     name: str = Field(examples=["MyGrid"])
+    """The name of the grid representing the geometry being linked to."""
+
     relative_path: str = Field(examples=["some/relative/path/mygrid.roff"])
+    """The relative path the grid on disk."""
```
Comment (tnatt) [inline src/fmu/dataio/datastructure/meta/content.py:133]:
```suggestion
    """The relative path to the grid on disk."""
```

File: src/fmu/dataio/datastructure/meta/content.py (hunk 1 above)
Comment (mferrera) [reply src/fmu/dataio/datastructure/meta/content.py]: I guess we don't want to store the emotional state of the data...

**Review by tnatt — APPROVED** [review]
LGTM 🙂

**Comment by tnatt** [comment]
I really appreciate your work on documentation btw 🥇

**Review by jcrivenaes — APPROVED** [review]
👍

**Pull request merged**